phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -648,6 +649,19 @@ public void clearTableRegionCache(byte[] tableName) throws SQLException {
         connection.clearRegionCache(TableName.valueOf(tableName));
     }
 
+    public byte[] getNextRegionStartKey(HRegionLocation regionLocation, byte[] currentKey) throws IOException {
```

Comment:
Are we sure this is the inconsitency?  I'd rather do something more protective where we compare the successive keys?

Reply:
updated the logic and handled the start and end empty row case